File: Angular/src/app/models/mentor.ts
import { Category } from "./category";
export class Mentor {
    employeeId: number = 0;
    firstName: string = '';
    lastName: string = '';
    email: string = '';
    password: string = '';
    jobTitle: string = '';
    company: string = '';
    location: string = '';
    bio: string = '';
    category: Category;
    skills: string[] = [];
    profileImage: number[] = []; // Array of bytes for storing the image as a byte array

    constructor(
        employeeId?: number,
        firstName?: string,
        lastName?: string,
        email?: string,
        password?: string,
        jobTitle?: string,
        company?: string,
        location?: string,
        bio?: string,
        category?: Object,
        skills?: string[],
        profileImage?: number[] // Initialize as an empty array
    ) {
        this.employeeId = employeeId || 0;
        this.firstName = firstName || '';
        this.lastName = lastName || '';
        this.email = email || '';
        this.password = password || '';
        this.jobTitle = jobTitle || '';
        this.company = company || '';
        this.location= location || '';
        this.bio = bio || '';
        this.category = new Category();
        this.skills = skills || [];
        this.profileImage = profileImage || [];
    }
}

File: Mentor/talent-share-portal-1/src/main/java/com/talentshare/mentor/model/Mentor.java
package com.talentshare.mentor.model;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.talentshare.mentor.constants.MessageConstants;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "mentorDetails")
public class Mentor {
	
	
	@Indexed(unique=true)
	//@NotNull(message = "Field must not be null")
	@Min(value = 1, message = MessageConstants.EMPLOYEE_ID_VALIDATION_MESSASGE)
	private int employeeId;
	
	@NotEmpty(message = "FirstName is required")
	@Pattern(regexp = "^[a-zA-Z]+$", message = MessageConstants.FIRSTNAME_VALIDATION_MESSASGE)
	private String firstName;
	
	@NotEmpty(message = "LastName is required")
	@Pattern(regexp = "^[a-zA-Z]+$", message = MessageConstants.LASTNAME_VALIDATION_MESSASGE)
	private String lastName;
	
	@Indexed(unique=true)
	@NotEmpty
	@Pattern(regexp = "^[A-Za-z0-9+_.-]+@(.+)$", message = MessageConstants.EMAIL_VALIDATION_MESSAGE)
	private String email;
	
	@JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
	private String password;
	
	@NotEmpty
	private String jobTitle;
	
	@NotEmpty
	private String location;
	
	@NotEmpty
	private String company;
	
	private Category category;
	
	@NotEmpty
	private List<String> skills;
	
	@NotEmpty
	private String bio;
	private LocalDateTime createdAt;
	private boolean isActive;
	private byte[] profileImage;
	
}


